pkg/mcshub/controller/controller_test.go

```diff
@@ -97,411 +82,291 @@ func TestImportGenerated(t *testing.T) {
-func TestUpdateExportAndImport(t *testing.T) {
-	exp, es := prepareServiceExport(t)
-	preloadedObjects := []runtime.Object{service, exp}
+func TestExportConflict(t *testing.T) {
+	assert := require.New(t)
+	exp1, err := prepareServiceExport(export, cluster1)
+	assert.Nil(err)
+	// change the port number on the service
+	exp2, err := prepareServiceExport(export, cluster2)
+	assert.Nil(err)
+	exp2.CreationTimestamp.Add(1 * time.Second)
+	spec := &mcs.ExportSpec{}
+	err = spec.UnmarshalObjectMeta(&exp2.ObjectMeta)
+	assert.Nil(err)
+	spec.Service.Ports[0].Port = 8080
+	err = spec.MarshalObjectMeta(&exp2.ObjectMeta)
+	assert.Nil(err)
+
+	preloadedObjects := []runtime.Object{exp1, exp2}
 	ser := controller.ServiceExportReconciler{
 		Client: getClient(preloadedObjects),
 		Log:    newLogger(t, false),
 		Scheme: getScheme(),
 	}
-	req := reconcile.Request{
+
+	_, err = ser.Reconcile(context.TODO(), reconcile.Request{
 		NamespacedName: types.NamespacedName{
-			Name:      exp.GetName(),
-			Namespace: exp.GetNamespace(),
-		}}
-	result, err := ser.Reconcile(context.TODO(), req)
-	if err != nil {
-		t.Error(err)
-	}
-	if result.Requeue {
-		t.Error("unexpected requeue")
-	}
+			Name:      exp1.GetName(),
+			Namespace: exp1.GetNamespace(),
+		}})
+	assert.Nil(err)
+	_, err = ser.Reconcile(context.TODO(), reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      exp2.GetName(),
+			Namespace: exp2.GetNamespace(),
+		}})
+	assert.Nil(err)
 
-	si := mcsv1a1.ServiceImport{}
-	err = ser.Client.Get(context.TODO(), req.NamespacedName, &si)
-	if err != nil {
-		t.Error(err)
-	}
-	// make sure that the right serviceImport was created
-	excepted := generateExceptedServiceImport(exp, es)
-	assertions := require.New(t)
-	compareSi(&excepted, &si, assertions)
+	imports := &mcsv1a1.ServiceImportList{}
+	err = ser.Client.List(context.TODO(), imports, client.InNamespace(brokerNS))
+	assert.Nil(err)
 
-	// update the export:
-	es.Service.Ports = append(es.Service.Ports, differentPort)
-	err = es.MarshalObjectMeta(&exp.ObjectMeta)
-	if err != nil {
-		t.Error(err)
-	}
+	assert.Len(imports.Items, 1, "expected 1 import got", len(imports.Items))
 
-	err = ser.Client.Update(context.TODO(), exp)
-	if err != nil {
-		//fails here - "Operation cannot be fulfilled on serviceexports.multicluster.x-k8s.io "svc": object was modified"
-		t.Error(err)
-	}
+	exports := &mcsv1a1.ServiceExportList{}
+	err = ser.Client.List(context.TODO(), exports, client.InNamespace(brokerNS))
+	assert.Nil(err)
 
-	result, err = ser.Reconcile(context.TODO(), req)
-	if err != nil {
-		t.Error(err)
-	}
-	if result.Requeue {
-		t.Error("unexpected requeue")
+	for _, exp := range exports.Items {
+		conflict := lhutil.GetServiceExportCondition(&exp.Status, mcsv1a1.ServiceExportConflict)
+		assert.NotNil(conflict)
+		assert.Equal(exp.GetName() == exp1.GetName(), conflict.Status == corev1.ConditionFalse,
+			"conflict set incorrectly", exp.GetName(), conflict)
```

Comment:
i think its also important to test conflict resolution when one of the exports is updated or deleted.
also test the conflict status is restored if spoke's export uploader update export (and delete all conditions for example)

Reply:
add conflict resolution in latest commit